contrib/terraform/packet/README.md

```diff
@@ -0,0 +1,231 @@
+# Kubernetes on Packet with Terraform
+
+Provision a Kubernetes cluster with [Terraform](https://www.terraform.io) on
+[Packet](https://www.packet.com).
+
+## Status
+
+This will install a Kubernetes cluster on Packet bare metal. It should work in all locations and on most hardware.
+
+## Approach
+The terraform configuration inspects variables found in
+[variables.tf](variables.tf) to create resources in your Packet project.
+There is a [python script](../terraform.py) that reads the generated`.tfstate`
+file to generate a dynamic inventory that is consumed by [cluster.yml](../../..//cluster.yml)
+to actually install Kubernetes with Kubespray.
+
+### Kubernetes Nodes
+You can create many different kubernetes topologies by setting the number of
+different classes of hosts.
+- Master nodes with etcd: `number_of_k8s_masters` variable
+- Master nodes without etcd: `number_of_k8s_masters_no_etcd` variable
+- Standalone etcd hosts: `number_of_etcd` variable
+- Kubernetes worker nodes: `number_of_k8s_nodes` variable
+
+Note that the Ansible script will report an invalid configuration if you wind up
+with an *even number* of etcd instances since that is not a valid configuration. This
+restriction includes standalone etcd nodes that are deployed in a cluster along with
+master nodes with etcd replicas. As an example, if you have three master nodes with
+etcd replicas and three standalone etcd nodes, the script will fail since there are
+now six total etcd replicas.
+
+## Requirements
+
+- [Install Terraform](https://www.terraform.io/intro/getting-started/install.html)
+- [Install Ansible](http://docs.ansible.com/ansible/latest/intro_installation.html)
+- you already have a Packet API key
+- you have an SSH key pair
+
+## Terraform
+Terraform will be used to provision all of the PAcket resources with base software as appropriate.
+
+### Configuration
+
+#### Inventory files
+
+Create an inventory directory for your cluster by copying the existing sample and linking the `hosts` script (used to build the inventory based on Terraform state):
+
+```ShellSession
+$ cp -LRp contrib/terraform/packet/sample-inventory inventory/$CLUSTER
+$ cd inventory/$CLUSTER
+$ ln -s ../../contrib/terraform/packet/hosts
```

Comment:
So the host file doesn't exist when this command is run. Is that correct? Might want to just add a note saying that the file gets created later (if that's correct).